tests/unit/test.index.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+var path = require('path');
+
+var home = path.join(__dirname, 'data', process.platform);
+process.env.HOME = home;
+process.env.USERPROFILE = home;
+
+var findSimulators = require('../../index');
+var platform = require('../../lib/platform');
+
+module.exports = {
+
+  'findSimulators works on current platform': function(test) {
+    var currentPlatform = platform(process.platform);
+    var vars = {
+      PROFILEDIR: currentPlatform.firefoxProfilesDir,
+      BINARY: currentPlatform.simulatorBinary
+    };
+
+    var expected = [
+      { version: '1.3',
+        bin: '%PROFILEDIR%/asdflulz.Release/extensions/fxos_1_3_simulator@mozilla.org/%BINARY%',
+        profile: '%PROFILEDIR%/asdflulz.Release/extensions/fxos_1_3_simulator@mozilla.org/profile' },
+      { version: '2.0',
+        bin: '%PROFILEDIR%/asdflulz.Release/extensions/fxos_2_0_simulator@mozilla.org/%BINARY%',
+        profile: '%PROFILEDIR%/asdflulz.Release/extensions/fxos_2_0_simulator@mozilla.org/profile' },
+      { version: '1.4',
+        bin: '%PROFILEDIR%/omgbbqya.dev-edition-default/extensions/fxos_1_4_simulator@mozilla.org/%BINARY%',
+        profile: '%PROFILEDIR%/omgbbqya.dev-edition-default/extensions/fxos_1_4_simulator@mozilla.org/profile' },
+      { version: '2.1',
+        bin: '%PROFILEDIR%/omgbbqya.dev-edition-default/extensions/fxos_2_1_simulator@mozilla.org/%BINARY%',
+        profile: '%PROFILEDIR%/omgbbqya.dev-edition-default/extensions/fxos_2_1_simulator@mozilla.org/profile' },
+      { version: '1.5',
+        bin: '%PROFILEDIR%/ou812wow.Nightly/extensions/fxos_1_5_simulator@mozilla.org/%BINARY%',
+        profile: '%PROFILEDIR%/ou812wow.Nightly/extensions/fxos_1_5_simulator@mozilla.org/profile' },
+      { version: '2.2',
+        bin: '%PROFILEDIR%/ou812wow.Nightly/extensions/fxos_2_2_simulator@mozilla.org/%BINARY%',
+        profile: '%PROFILEDIR%/ou812wow.Nightly/extensions/fxos_2_2_simulator@mozilla.org/profile' }
+    ];
+
+    // Replace platform-specific placeholders in the expected data
+    for (var idx = 0; idx < expected.length; idx++) {
```

Comment:
Oh god the two/three letter variables. Could they be more verbose? I find a lot of short iterators hard to read. But then I am an old man.

![upf1](https://cloud.githubusercontent.com/assets/90871/5928012/5d353854-a645-11e4-8c78-a8f891e37832.gif)